app/assets/javascripts/discourse-common/addon/utils/dom-utils.js

```diff
@@ -0,0 +1,51 @@
+function isWindow(obj) {
+  return obj != null && obj === window;
+}
+
+function scrollTop(element) {
+  return element
+    ? element.scrollTop
+    : window.pageYOffset || document.documentElement.scrollTop;
+}
+
+function offset(element) {
+  // note that getBoundingClientRect forces a reflow.
+  // When used in critical performance conditions
+  // you might want to move to more involved solution
+  // such as implementing an IntersectionObserver and
+  // using its boundingClientRect property
+  const rect = element.getBoundingClientRect();
+  return {
+    top: rect.top + window.scrollY,
+    left: rect.left + window.scrollX,
+  };
+}
+
+function height(element) {
```

Comment:
For readability, I think I prefer using `window.innerHeight` and `element.clientHeight` directly. (Same for width.) I don't see an advantage to abstracting this under the `domUtils` helper.

Reply:
Yes window is probably enough, I just wanted it to be a no brainer